flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSource.java

```diff
@@ -247,6 +258,24 @@ int inferParallelism(FlinkInputFormat format, ScanContext context) {
       parallelism = Math.max(1, parallelism);
       return parallelism;
     }
+
+    boolean localityEnabled() {
+      FileIO fileIO = table.io();
+      if (fileIO instanceof HadoopFileIO) {
+        Boolean localityConfig = readableConfig.get(FlinkConfigOptions.TABLE_EXEC_ICEBERG_EXPOSE_SPLIT_LOCALITY_INFO);
+
+        HadoopFileIO hadoopFileIO = (HadoopFileIO) fileIO;
+        try {
+          String scheme = new Path(table.location()).getFileSystem(hadoopFileIO.getConf()).getScheme();
+          boolean defaultValue = FILE_SYSTEM_SUPPORT_LOCALITY.contains(scheme);
```

Comment:
It would be nice to avoid the dependency on `Path` in this class - so people who don't have hadoop needs don't need the dependency, you might be able to do the following to get the same result.

If you call `((HadoopFileIO) fileIO).newInputFile(table.location()).location()`, I believe you can check the scheme on the location sting returned by `InputFile`.

There could very possibly be.a cleaner way to do it. But if this class doesn't already pull in `org.apache.hadoop.*`, then it would be best to avoid that if possible so that users who run only on cloud environments for example don't have to have HDFS on their classpath.

In this case, just be having the import, even if it's `S3FileIO`, users still need the hadoop jars on the classpath (which is sometimes a pain on Flink).

Reply:
Because fileio is an instance of HadoopFileIO, when we call the `HadoopFileIO#newInputFile` method, does this still not addressed? It still depends on `org.apache.hadoop.`

https://github.com/apache/iceberg/blob/80fd7fd4f0ec365f3d002aaae40329bb7eed77d4/core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java#L57-L61